src/qililab/platform/platform.py

```diff
@@ -627,18 +635,30 @@ def execute_anneal_program(
             transpiler (Callable): ising to flux transpiler. The transpiler should take 2 values as arguments (delta, epsilon) and return 2 values (phix, phiz)
             averages (int, optional): Amount of times to run and average the program over. Defaults to 1.
         """
-        annealing_program = AnnealingProgram(self, annealing_program_dict)
-        annealing_program.transpile(transpiler)
-        annealing_waveforms = annealing_program.get_waveforms()
-
-        qp_annealing = QProgram()
-        with qp_annealing.average(averages):
-            for bus, waveform in annealing_waveforms.values():
-                qp_annealing.play(bus=bus.alias, waveform=waveform)
-
-        # TODO: define readout
-
-        self.execute_qprogram(qprogram=qp_annealing)
+        if calibration and calibration.has_waveform(bus=readout_bus, name=measurement_name):
+            annealing_program = AnnealingProgram(self, annealing_program_dict)
+            annealing_program.transpile(transpiler)
+            annealing_waveforms = annealing_program.get_waveforms()
+
+            qp_annealing = QProgram()
+            with qp_annealing.average(averages):
+                for bus, waveform in annealing_waveforms.values():
+                    qp_annealing.play(bus=bus.alias, waveform=waveform)
+
+                    if weights and calibration.has_weights(bus=readout_bus, name=weights):
+                        qp_annealing.measure(bus=readout_bus, waveform=measurement_name, weights=weights)
+                    else:
+                        r_duration = calibration.get_waveform(bus=readout_bus, name=measurement_name).get_duration()
+                        weights_shape = Square(amplitude=1, duration=r_duration)
+                        qp_annealing.measure(
+                            bus=readout_bus, waveform=measurement_name, weights=IQPair(I=weights_shape, Q=weights_shape)
+                        )
+
+            self.execute_qprogram(qprogram=qp_annealing, calibration=calibration)
```

Comment:
shouldn't the method return the results of the `execute_qprogram` ?

Reply:
You are totally right, we missed that in Victor's PR and I totally missed it too while coding this. Thanks.

Reply:
Yup i didnt add it because I didnt add measurements, that's why i left the TODO: add measurements solved in this pr :P

Reply:
Adding it, thanks.